src/types/types.ts

```diff
@@ -75,23 +89,33 @@ export type Organization = {
   phoneNumber: VerifiedTextField;
 };
 
-const DEFAULT_TEXT_FIELD: VerifiedTextField = {
-  value: "",
-  verified: false,
-};
-
 export const DEFAULT_ORGANIZATION: Organization = {
   name: DEFAULT_TEXT_FIELD,
   address: DEFAULT_TEXT_FIELD,
   website: DEFAULT_TEXT_FIELD,
   phoneNumber: DEFAULT_TEXT_FIELD,
 };
 
-export const isVerified = <T extends Record<string, VerifiedField>>(
-  fields: T,
-  verified: boolean = true,
-): boolean =>
-  fields && Object.values(fields).every((field) => field.verified === verified);
+export const isVerified = <T>(data: T, verified: boolean = true): boolean => {
+  if (typeof data !== "object" || !data) return false;
+
+  return Object.values(data).every((field) => {
+    if (Array.isArray(field)) {
+      return field.every(
+        (item) => typeof item === "object" && item.verified === verified,
+      );
+    }
+
+    if (typeof field === "object" && field !== null) {
+      if ("verified" in field) {
+        return field.verified === verified;
+      }
+      return isVerified(field, verified);
+    }
+
+    return true;
+  });
+};
 
```

Comment:
Should do the falsy check before the type check

`if (typeof data !== "object" || !data) return false;`
`if (typeof field === "object" && field !== null)`

also != null is better because it covers both null and undefined

Also better to move the null check to cover both if statements
```JavaScript
export const isVerified = <T>(data: T, verified: boolean = true): boolean => {
  if (data == null || typeof data !== "object") return false;

  return Object.values(data).every((field) => {
    if (field != null) {
      if (Array.isArray(field)) {
        return field.every(
          (item) => typeof item === "object" && item.verified === verified,
        );
      }

      if (typeof field === "object") {
        if ("verified" in field) {
          return field.verified === verified;
        }
        return isVerified(field, verified);
      }
    }

    return true;
  });
};
```

The recursion on 113 is a bit unmaintainable.

This function probably needs a rewrite, define interfaces for what is being verified instead of doing a recursive search for some field. Then specify those acceptable types instead of a generic T .

Also why do we have a function in a type definition file? Should be in some common functions module

Reply:
I now have functions to check separately verifiable field records and arrays to avoid the recursive loop.